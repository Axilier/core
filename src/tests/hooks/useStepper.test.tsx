import React from 'react';
import { mount } from 'enzyme';
import { useStepper } from '../../index';

// region demo
const Demo = (): JSX.Element => {
    const [state, increment, decrement, zero] = useStepper(1);
    return (
        <div id={'container'}>
            <div id={'state'}>{state}</div>
            <button type={'button'} id={'increment'} onClick={increment}>
                increment
            </button>
            <button type={'button'} id={'decrement'} onClick={decrement}>
                decrement
            </button>
            <button type={'button'} id={'zero'} onClick={zero}>
                zero
            </button>
        </div>
    );
};
// endregion

const demo = mount(<Demo />);

describe('when I render the demo in an application', () => {
    it('should render correctly', () => {
        expect(demo).toMatchSnapshot();
    });
    it('should initially have a state of zero', () => {
        expect(demo.find('#state').text()).toBe('1');
    });
    describe('when I click the increment button', () => {
        beforeAll(() => demo.find('#increment').simulate('click'));
        it('should increment the state by 1', () => {
            expect(demo.find('#state').text()).toBe('2');
        });
    });
    describe('when I click the decrement button', () => {
        beforeAll(() => demo.find('#decrement').simulate('click'));
        it('should decrement the state by 1', () => {
            expect(demo.find('#state').text()).toBe('1');
        });
    });
    describe('when I click the zero button', () => {
        beforeAll(() => demo.find('#zero').simulate('click'));
        it('should set the state to zero', () => {
            expect(demo.find('#state').text()).toBe('0');
        });
    });
});
