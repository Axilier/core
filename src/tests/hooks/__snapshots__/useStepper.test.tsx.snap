// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when I render the demo in an application should render correctly 1`] = `
<Demo>
  <div
    id="container"
  >
    <div
      id="state"
    >
      1
    </div>
    <button
      id="increment"
      onClick={[Function]}
      type="button"
    >
      increment
    </button>
    <button
      id="decrement"
      onClick={[Function]}
      type="button"
    >
      decrement
    </button>
    <button
      id="zero"
      onClick={[Function]}
      type="button"
    >
      zero
    </button>
  </div>
</Demo>
`;
